### PR TITLE
Support reading bytes by offset

### LINKED ISSUE #127: Feature: Support reading bytes by offset
[Restic](https://github.com/restic/restic) is a backup program. We would like to implement generic cloud storage backend for restic using Stow. Restic needs a way to read bytes from a file for a given offset and length.

https://github.com/restic/restic/issues/830#issuecomment-282974191

What do you think about adding a method file `Partial()` similar to `Open()` for this?

### PR DESCRIPTION
**Pull request by tamalsaha**
Fixes #127

### CODE REVIEW

**Comment by piotrrojek** [comment]
Looks awesome! The only thing missing is some test for all the adaptors in `test/test.go`.. :) would you mind adding one?

**Comment by TamalSaha** [comment]
@piotrrojek I will add tests. But first I want to finish implementing restic backend and test things manually.

File: local/item.go [hunk 1]
```diff
@@ -70,6 +72,48 @@ func (i *item) Open() (io.ReadCloser, error) {
 	return os.Open(i.path)
 }
 
+// LimitedReadCloser wraps io.LimitedReader and exposes the Close() method.
+type LimitedReadCloser struct {
```
Comment (matryer) [inline local/item.go]: It is enough to include `io.ReadCloser` - don't need to specify `io.Reader` too.

Comment (matryer) [inline local/item.go]: Can we un-export this type? The method returning `io.ReadCloser` it's probably enough. We can keep the type private?

Comment (TamalSaha) [inline local/item.go]: Can we un-export this type? The method returning io.ReadCloser it's probably enough. We can keep the type private? - yes.

File: azure/item.go [hunk 2]
```diff
@@ -47,6 +48,23 @@ func (i *item) Open() (io.ReadCloser, error) {
 	return i.client.GetBlob(i.container.id, i.id)
 }
 
+func (i *item) Partial(length, offset int64) (io.ReadCloser, error) {
```
Comment (matryer) [inline azure/item.go]: Thinking out loud here... I wonder if we could just implement `io.ReadSeeker`?

Comment (TamalSaha) [inline azure/item.go]: For my use-case, I need `io.ReadCloser`. I am trying to use Stow to allow backup to any cloud storage. https://github.com/restic/restic/pull/885

Comment (matryer) [inline azure/item.go]: Why don't we return a `ReadSeekCloser`? It'll feel familiar.

**Comment by tamalsaha** [comment]
@matryer , sorry for being quiet on this thread. My original motivation for adding this method were so that I could use that with restic. But restic project has decided to use the underlying libraries directly. So, we don't this pr anymore.

**Pull request closed without merging**